web/src/components/overview/ProductSection.jsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import { Text } from "@patternfly/react-core";
+import { toValidationError, useCancellablePromise } from "~/utils";
+import { useInstallerClient } from "~/context/installer";
+import { useProduct } from "~/context/product";
+import { If, Section, SectionSkeleton } from "~/components/core";
+import { _ } from "~/i18n";
+
+const errorsFrom = (issues) => {
+  const errors = issues.filter(i => i.severity === "error");
+  return errors.map(toValidationError);
+};
+
+export default function ProductSection() {
+  const { software } = useInstallerClient();
+  const [issues, setIssues] = useState([]);
+  const { selectedProduct } = useProduct();
+  const { cancellablePromise } = useCancellablePromise();
+
+  useEffect(() => {
+    cancellablePromise(software.product.getIssues()).then(setIssues);
+    return software.product.onIssuesChange(setIssues);
+  }, [cancellablePromise, setIssues, software]);
+
+  const Content = ({ isLoading = false }) => {
+    return (
+      <If
+        condition={isLoading}
+        then={<SectionSkeleton numRows={1} />}
+        else={
+          <Text>
+            {selectedProduct?.name}
+          </Text>
+        }
+      />
+    );
+  };
```

Comment:
Same than in previous comment. Since you're already passing props to this internal component, I think it's better to pass `productName` prop too and move it outside. 

About the use of `<If>`, I think it's more readable/appropriated to use an early `return` instead. 

```suggestion
  const Content = ({ isLoading = false }) => {
    if (isLoading) return <SectionSkeleton numRows={1} />
    
    return <Text>{selectedProduct?.name}</Text>;
  };
```

I can hear your complaining. Right, it was me who proposed the use of the `<If>` component, but more for avoiding to interpolate expressions like `{ isWhatever && <RenderThis /> }` in the middle of a JSX piece.